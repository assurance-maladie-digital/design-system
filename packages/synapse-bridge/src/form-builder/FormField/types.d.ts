export interface PeriodValue {
	from: string | null;
	to: string | null;
}

export interface RangeValue {
	min: number | null;
	max: number | null;
}

export type FieldItemValue = string | number | null | undefined;

export type OtherFieldValue = string | null;

export type ChoiceFieldValue = FieldItemValue | FieldItemValue[];

export interface ChoiceValue {
	value: ChoiceFieldValue;
	other?: OtherFieldValue;
}

export type FieldValue = string | number | null | undefined | PeriodValue | ChoiceValue | number[];

export type ErrorMessages = string | string[];

export interface ChoiceFieldErrorMessages {
	value?: ErrorMessages;
	other?: ErrorMessages;
}

export interface PeriodErrorMessages {
	from?: ErrorMessages;
	to?: ErrorMessages;
}

export interface FieldItem {
	/**
	 * @deprecated Use `title` instead.
	 */
	text?: string;
	title?: string;
	value: FieldItemValue;
	alone?: boolean;
}

interface FieldItemNew {
	title: string;
	value: FieldItemValue;
	alone?: boolean;
}

interface FieldItemDeprecated {
	/**
	 * @deprecated Use `title` instead.
	 */
	text: string;
	value: FieldItemValue;
	alone?: boolean;
}

export type FieldItem = FieldItemNew | FieldItemDeprecated;

export interface FieldOptions {
	errorMessages?: ErrorMessages | ChoiceFieldErrorMessages | PeriodErrorMessages;
	[key: string]: unknown;
}

export interface OtherField {
	label?: string;
	fieldOptions?: FieldOptions;
	selectedChoice?: FieldItemValue;
}

export interface Field {
	type: string;
	value?: FieldValue;
	title?: string;
	min?: number;
	max?: number;
	description?: string;
	tooltip?: string;
	multiple?: boolean;
	fieldOptions?: FieldOptions;
	items?: FieldItem[];
	other?: OtherField | OtherField[];
	mask?: string;
	dynamic?: boolean;
}
