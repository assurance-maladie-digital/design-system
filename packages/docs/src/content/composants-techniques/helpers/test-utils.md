---
title: testUtils
description: Le helper `testUtils` contient un ensemble de fonctions utilitaires pour les tests unitaires.
---

<doc-tabs>

<doc-tab-item label="Utilisation">

<doc-alert type="info">

Le helper `testUtils` exporte également toutes les fonctions du package [Vue Test Utils](https://vue-test-utils.vuejs.org/).

</doc-alert>

### applyGlobalConfig

La fonction `applyGlobalConfig` permet d’appliquer la configuration globale pour les tests, elle est utilisée dans le fichier `setup.ts`.

```ts
import { applyGlobalConfig } from '@cnamts/vue-dot/src/helpers/testUtils';

applyGlobalConfig();
```

### installGlobalPlugins

La fonction `installGlobalPlugins` permet d’installer les plugins [Vue Input Facade](https://ronaldjerez.github.io/vue-input-facade/), [Vue Meta](https://vue-meta.nuxtjs.org/), [Vuex](https://vuex.vuejs.org/), [Vuetify](https://v2.vuetifyjs.com/) et Vue Dot pour les tests.

```ts
import { createLocalVue, installGlobalPlugins } from '@cnamts/vue-dot/src/helpers/testUtils';

const localVue = createLocalVue();

installGlobalPlugins(localVue);
```

### installRouter

La fonction `installRouter` permet d’installer [Vue Router](https://v3.router.vuejs.org/fr/) pour les tests.

```ts
import { createLocalVue, installRouter } from '@cnamts/vue-dot/src/helpers/testUtils';

const localVue = createLocalVue();

installRouter(localVue);
```

### createRouter

La fonction `createRouter` permet de créer une instance de [Vue Router](https://v3.router.vuejs.org/fr/) pour les tests.

```ts
import { createRouter } from '@cnamts/vue-dot/src/helpers/testUtils';

const routes = [];

const router = createRouter({
	routes
});
```

### createStore

La fonction `createStore` permet de créer une instance de [Vuex](https://vuex.vuejs.org/) pour les tests.

```ts
import { createStore } from '@cnamts/vue-dot/src/helpers/testUtils';

const store = createStore({
	state: {}
});
```

### createVuetifyInstance

La fonction `createVuetifyInstance` permet de créer une instance de [Vuetify](https://v2.vuetifyjs.com/) pour les tests.

```ts
import { createVuetifyInstance } from '@cnamts/vue-dot/src/helpers/testUtils';

const vuetify = createVuetifyInstance();
```

### mockVFormRef

La fonction `mockVFormRef` permet de mocker la référence d’un composant [VForm](https://v2.vuetifyjs.com/en/components/forms/) pour les tests.

```ts
import { mockVFormRef } from '@cnamts/vue-dot/src/helpers/testUtils';

const mocks = {
	$refs: {
		form: mockVFormRef(true)
	}
};
```

### mockTranslations

La fonction `mockTranslations` permet de mocker les traductions pour les tests.

```ts
import { mockTranslations } from '@cnamts/vue-dot/src/helpers/testUtils';

const mocks = {
	...mockTranslations<string | string[]>({
		'path.to.title': 'Titre',
		'path.to.items': [
			'Élément 1',
			'Élément 2',
			'Élément 3'
		]
	})
};
```

</doc-tab-item>

<doc-tab-item label="API">
<doc-api name="helpers/test-utils"></doc-api>
</doc-tab-item>

</doc-tabs>
