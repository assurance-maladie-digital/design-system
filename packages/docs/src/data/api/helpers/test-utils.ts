import { Api } from '~/types';

export const api: Api = {
	applyGlobalConfig: {
		returnValue: [
			{
				type: 'void',
			},
		],
	},
	installGlobalPlugins: {
		arguments: [
			{
				name: 'localVue',
				description: 'Local Vue instance',
				type: 'Vue',
				required: true,
			},
		],
		returnValue: [
			{
				type: 'void',
			},
		],
	},
	installRouter: {
		arguments: [
			{
				name: 'localVue',
				description: 'Local Vue instance',
				type: 'Vue',
				required: true,
			},
		],
		returnValue: [
			{
				type: 'void',
			},
		],
	},
	createRouter: {
		arguments: [
			{
				name: 'options',
				description: 'Router options',
				type: 'RouterOptions',
				required: false,
			},
		],
		returnValue: [
			{
				type: 'VueRouter',
			},
		],
	},
	createStore: {
		arguments: [
			{
				name: 'options',
				description: 'Store options',
				type: 'StoreOptions',
				required: false,
			},
		],
		returnValue: [
			{
				type: 'Store',
			},
		],
	},
	createVuetifyInstance: {
		arguments: [
			{
				name: 'preset',
				description: 'Vuetify preset',
				type: 'Partial<UserVuetifyPreset>',
				required: true,
			},
		],
		returnValue: [
			{
				type: 'Vuetify',
			},
		],
	},
	mockVFormRef: {
		arguments: [
			{
				name: 'isValid',
				description: 'Is valid',
				type: 'boolean',
				required: true,
			},
		],
		returnValue: [
			{
				type: 'void',
			},
		],
	},
	mockTranslations: {
		arguments: [
			{
				name: 'translations',
				description: 'Translations',
				type: 'IndexedObject',
				required: true,
			},
		],
		returnValue: [
			{
				type: 'void',
			},
		],
	},
};
