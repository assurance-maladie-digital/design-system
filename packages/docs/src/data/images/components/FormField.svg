<svg width="541" height="357" viewBox="0 0 541 357" fill="none" xmlns="http://www.w3.org/2000/svg">
<rect width="541" height="357" fill="#E7ECF5"/>
<rect x="63" y="88" width="415" height="181" fill="white"/>
<rect x="171" y="151" width="200" height="56" rx="4" fill="white"/>
<rect x="171.5" y="151.5" width="199" height="55" rx="3.5" stroke="black" stroke-opacity="0.38"/>
<rect opacity="0.5" x="187" y="170" width="136" height="19" rx="9.5" fill="#CED9EB"/>
</svg>
