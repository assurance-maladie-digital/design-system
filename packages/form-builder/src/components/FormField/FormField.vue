<template>
	<div class="vd-form-field">
		<h4
			v-if="field.title"
			class="text-body-1"
			:class="{ 'mb-2': !field.tooltip }"
		>
			<span v-html="field.title" />

			<VTooltip
				v-if="field.tooltip"
				right
			>
				<template #activator="{ on }">
					<VBtn
						icon
						class="ml-1"
						v-on="on"
					>
						<VIcon size="20">
							{{ informationIcon }}
						</VIcon>
					</VBtn>
				</template>

				<span
					class="vd-form-field-tooltip-text"
					v-text="field.tooltip"
				/>
			</VTooltip>
		</h4>

		<p
			v-if="field.description"
			class="text-body-2 grey--text"
		>
			{{ field.description }}
		</p>

		<component
			:is="fieldComponent"
			v-if="isDefinedField"
			:field="field"
			@change="emitChangeEvent"
		/>

		<slot
			v-else
			:name="field.type"
			v-bind="{
				field,
				emitChangeEvent
			}"
		/>
	</div>
</template>

<script lang="ts">
	import Vue, { PropType } from 'vue';
	import Component, { mixins } from 'vue-class-component';

	import { Field } from './types';

	import { FieldMap } from './mixins/fieldMap';

	import { mdiInformationOutline } from '@mdi/js';

	const Props = Vue.extend({
		props: {
			field: {
				type: Object as PropType<Field>,
				required: true
			}
		}
	});

	const MixinsDeclaration = mixins(Props, FieldMap);

	@Component({
		model: {
			prop: 'field',
			event: 'change'
		}
	})
	export default class FormField extends MixinsDeclaration {
		informationIcon = mdiInformationOutline;

		get fieldComponent(): string {
			let fieldType = this.field.type;

			// Handle subtypes that are not in type 'select'
			if (fieldType !== 'select') {
				const fieldOptionsType = this.field.fieldOptions?.type as string || undefined;
				fieldType = fieldOptionsType || fieldType;
			}

			return this.getField(fieldType);
		}

		get isDefinedField(): boolean {
			return this.fieldExists(this.field.type);
		}

		emitChangeEvent(value: Field): void {
			this.$emit('change', value);
		}
	}
</script>

<style lang="scss" scoped>
	.vd-form-field + .vd-form-field {
		margin-top: 12px;
	}

	.vd-form-field-tooltip-text {
		white-space: pre-wrap;
	}
</style>
