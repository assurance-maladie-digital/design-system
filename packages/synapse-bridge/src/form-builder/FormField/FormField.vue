<script lang="ts">
import { defineComponent } from 'vue'
import type { PropType } from 'vue'

import type { Field } from './types'

import FieldMap from './mixins/fieldMap'

import { mdiInformationOutline } from '@mdi/js'

export default defineComponent({
	mixins: [FieldMap],
	props: {
		modelValue: {
			type: Object as PropType<Field>,
			required: true,
		},
	},
	emits: ['update:modelValue'],
	data() {
		return {
			informationIcon: mdiInformationOutline,
		}
	},
	computed: {
		fieldComponent(): string {
			let fieldType = this.modelValue.type

			// Handle subtypes that are not in type 'select'
			if (fieldType !== 'select') {
				const fieldOptionsType =
					(this.modelValue.fieldOptions?.type as string) || undefined
				fieldType = fieldOptionsType || fieldType
			}

			return this.getField(fieldType)
		},

		isDefinedField(): boolean {
			return this.fieldExists(this.modelValue.type)
		},
	},
	methods: {
		emitChangeEvent(value: Field): void {
			this.$emit('update:modelValue', value)
		},
	},
})
</script>

<template>
	<div class="vd-form-field">
		<h4
			v-if="modelValue.title"
			class="text-body-1 d-flex align-sm-center"
			:class="{ 'mb-2': !modelValue.tooltip }"
		>
			<span v-html="modelValue.title" />

			<VTooltip v-if="modelValue.tooltip" offset="50px">
				<template #activator="{ props }">
					<VBtn
						v-bind="props"
						class="ml-1"
						icon
						variant="flat"
						density="compact"
					>
						<VIcon size="20">
							{{ informationIcon }}
						</VIcon>
					</VBtn>
				</template>

				<span
					class="vd-form-field-tooltip-text"
					v-text="modelValue.tooltip"
				/>
			</VTooltip>
		</h4>

		<p v-if="modelValue.description" class="text-body-2 text-grey">
			{{ modelValue.description }}
		</p>

		<component
			:is="fieldComponent"
			v-if="isDefinedField"
			:field="modelValue"
			@update:modelValue="emitChangeEvent"
		/>

		<slot
			v-else
			:name="modelValue.type"
			v-bind="{
				modelValue,
				emitChangeEvent,
			}"
		/>
	</div>
</template>

<style lang="scss" scoped>
.vd-form-field + .vd-form-field {
	margin-top: 12px;
}

.vd-form-field-tooltip-text {
	white-space: pre-wrap;
}
</style>
