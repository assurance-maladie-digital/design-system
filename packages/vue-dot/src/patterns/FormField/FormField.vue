<template>
	<component
		:is="getFieldType()"
		v-model="field"
		class="vd-fieldGroup-field"
	/>
</template>

<script lang="ts">
	import Vue, { PropType } from 'vue';
	import Component, { mixins } from 'vue-class-component';

	import { Field } from './types';

	import { FieldMap } from './mixins/fieldMap';

	const Props = Vue.extend({
		props: {
			/** v-model value */
			value: {
				type: Object as PropType<Field>,
				required: true
			}
		}
	});

	const MixinsDeclaration = mixins(Props, FieldMap);

	/** FormField is a component used to display */
	@Component({
		model: {
			prop: 'value',
			event: 'change'
		}
	})
	export default class FormField extends MixinsDeclaration {
		/**
		 * Get the field type from metadata or default type
		 *
		 * @returns {string} The field type
		 */
		getFieldType(): string {
			const metadataType = this.field.metadata ? this.field.metadata.type as string : undefined;

			return this.getField(metadataType || this.field.type);
		}

		get field() {
			return this.value;
		}

		set field(value: Field) {
			this.$emit('change', value);
		}
	}
</script>

<style lang="scss" scoped>
	.vd-fieldGroup-field {
		width: 100%;
	}
</style>
