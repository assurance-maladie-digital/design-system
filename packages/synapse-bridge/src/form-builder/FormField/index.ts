import FormField from './FormField.vue';

export default FormField;
