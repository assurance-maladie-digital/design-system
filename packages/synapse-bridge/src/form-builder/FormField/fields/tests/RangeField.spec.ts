import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import { vuetify } from '@tests/unit/setup'
import FormField from '../../FormField.vue'

describe('ChoiceSliderField', () => {
	it('should render correctly', () => {
		const wrapper = mount(FormField, {
			props: {
				modelValue: {
					title: 'Combien de jours par semaine souhaitez-vous travailler ?',
					type: 'range',
					value: [0, 'should be removed', 6] as any,
					max: 6,
					min: 0,
				},
			},
			global: {
				plugins: [vuetify],
			},
		})

		expect(wrapper).toMatchSnapshot()
	})

	it('render correctly in multiple mode', () => {
		const wrapper = mount(FormField, {
			props: {
				modelValue: {
					title: 'Combien de jours par semaine souhaitez-vous travailler ?',
					type: 'range',
					value: null,
					max: 6,
					min: 0,
					fieldOptions: {
						color: 'red',
					},
				},
			},
			global: {
				plugins: [vuetify],
			},
		})

		expect(wrapper).toMatchSnapshot()
	})
})
