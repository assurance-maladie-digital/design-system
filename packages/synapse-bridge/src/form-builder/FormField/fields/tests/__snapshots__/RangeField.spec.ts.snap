// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ChoiceSliderField > render correctly in multiple mode 1`] = `
<div data-v-9067cb71="" class="vd-form-field">
  <h4 data-v-9067cb71="" class="text-body-1 d-flex align-sm-center mb-2"><span data-v-9067cb71="">Combien de jours par semaine souhaitez-vous travailler ?</span>
    <!--v-if-->
  </h4>
  <!--v-if-->
  <div data-v-9067cb71="" class="vd-range-field" color="red">
    <div class="d-flex flex-wrap max-width-none ma-n3">
      <div class="v-input v-input--horizontal v-input--center-affix v-input--density-default v-locale--is-ltr v-input--dirty v-text-field ma-3">
        <!---->
        <div class="v-input__control">
          <div class="v-field v-field--active v-field--center-affix v-field--dirty v-field--variant-filled v-theme--light v-locale--is-ltr">
            <div class="v-field__overlay"></div>
            <div class="v-field__loader">
              <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
                <!---->
                <div class="v-progress-linear__background" style="width: 100%;"></div>
                <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
                  <div class="v-progress-linear__indeterminate">
                    <div class="v-progress-linear__indeterminate long"></div>
                    <div class="v-progress-linear__indeterminate short"></div>
                  </div>
                </transition-stub>
                <!---->
              </div>
            </div>
            <!---->
            <div class="v-field__field" data-no-activator=""><label class="v-label v-field-label v-field-label--floating" aria-hidden="true" for="input-0">
                <!---->Valeur min
              </label><label class="v-label v-field-label" for="input-0">
                <!---->Valeur min
              </label>
              <!----><input size="1" type="number" id="input-0" aria-describedby="input-0-messages" inputmode="numeric" class="v-field__input">
              <!---->
            </div>
            <!---->
            <!---->
            <div class="v-field__outline">
              <!---->
              <!---->
            </div>
          </div>
        </div>
        <!---->
        <!---->
      </div>
      <div class="v-input v-input--horizontal v-input--center-affix v-input--density-default v-locale--is-ltr v-input--dirty v-text-field ma-3">
        <!---->
        <div class="v-input__control">
          <div class="v-field v-field--active v-field--center-affix v-field--dirty v-field--variant-filled v-theme--light v-locale--is-ltr">
            <div class="v-field__overlay"></div>
            <div class="v-field__loader">
              <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
                <!---->
                <div class="v-progress-linear__background" style="width: 100%;"></div>
                <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
                  <div class="v-progress-linear__indeterminate">
                    <div class="v-progress-linear__indeterminate long"></div>
                    <div class="v-progress-linear__indeterminate short"></div>
                  </div>
                </transition-stub>
                <!---->
              </div>
            </div>
            <!---->
            <div class="v-field__field" data-no-activator=""><label class="v-label v-field-label v-field-label--floating" aria-hidden="true" for="input-2">
                <!---->Valeur max
              </label><label class="v-label v-field-label" for="input-2">
                <!---->Valeur max
              </label>
              <!----><input size="1" type="number" id="input-2" aria-describedby="input-2-messages" ripple="false" inputmode="numeric" class="v-field__input">
              <!---->
            </div>
            <!---->
            <!---->
            <div class="v-field__outline">
              <!---->
              <!---->
            </div>
          </div>
        </div>
        <!---->
        <!---->
      </div>
    </div>
    <div class="v-input v-input--horizontal v-input--center-affix v-input--density-default v-locale--is-ltr v-input--dirty v-slider v-range-slider v-locale--is-ltr align-center mt-2 mb-6">
      <div class="v-input__prepend">
        <!---->0
        <!---->
      </div>
      <div class="v-input__control">
        <div class="v-slider__container"><input id="input-4_start" name="input-4" tabindex="-1"><input id="input-4_stop" name="input-4" tabindex="-1">
          <div class="v-slider-track" style="--v-slider-track-size: 4px; --v-slider-tick-size: 2px;">
            <div class="v-slider-track__background bg-primary v-slider-track__background--opacity" style="inset-inline-start: 0%; width: 100%;"></div>
            <div class="v-slider-track__fill bg-primary" style="inset-inline-start: 0%; width: 100%;"></div>
            <!---->
          </div>
          <div class="v-slider-thumb v-slider-thumb--focused v-locale--is-ltr" style="--v-slider-thumb-position: 0%; --v-slider-thumb-size: 20px;" role="slider" tabindex="0" aria-valuemin="0" aria-valuemax="6" aria-valuenow="0" aria-readonly="false" aria-orientation="horizontal" aria-describedby="input-4-messages">
            <div class="v-slider-thumb__surface text-primary elevation-2"></div>
            <div class="v-slider-thumb__ripple text-primary"></div>
            <transition-stub name="scale-transition" appear="false" persisted="false" css="true">
              <div class="v-slider-thumb__label-container" style="display: none;">
                <div class="v-slider-thumb__label">
                  <div>0</div>
                </div>
              </div>
            </transition-stub>
          </div>
          <div class="v-slider-thumb v-slider-thumb--focused v-locale--is-ltr" style="--v-slider-thumb-position: 100%; --v-slider-thumb-size: 20px;" role="slider" tabindex="0" aria-valuemin="0" aria-valuemax="6" aria-valuenow="6" aria-readonly="false" aria-orientation="horizontal" aria-describedby="input-4-messages">
            <div class="v-slider-thumb__surface text-primary elevation-2"></div>
            <div class="v-slider-thumb__ripple text-primary"></div>
            <transition-stub name="scale-transition" appear="false" persisted="false" css="true">
              <div class="v-slider-thumb__label-container" style="display: none;">
                <div class="v-slider-thumb__label">
                  <div>6</div>
                </div>
              </div>
            </transition-stub>
          </div>
        </div>
      </div>
      <div class="v-input__append">
        <!---->6
      </div>
      <!---->
    </div>
  </div>
</div>
`;

exports[`ChoiceSliderField > should render correctly 1`] = `
<div data-v-9067cb71="" class="vd-form-field">
  <h4 data-v-9067cb71="" class="text-body-1 d-flex align-sm-center mb-2"><span data-v-9067cb71="">Combien de jours par semaine souhaitez-vous travailler ?</span>
    <!--v-if-->
  </h4>
  <!--v-if-->
  <div data-v-9067cb71="" class="vd-range-field" color="primary">
    <div class="d-flex flex-wrap max-width-none ma-n3">
      <div class="v-input v-input--horizontal v-input--center-affix v-input--density-default v-locale--is-ltr v-input--dirty v-text-field ma-3">
        <!---->
        <div class="v-input__control">
          <div class="v-field v-field--active v-field--center-affix v-field--dirty v-field--variant-filled v-theme--light v-locale--is-ltr">
            <div class="v-field__overlay"></div>
            <div class="v-field__loader">
              <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
                <!---->
                <div class="v-progress-linear__background" style="width: 100%;"></div>
                <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
                  <div class="v-progress-linear__indeterminate">
                    <div class="v-progress-linear__indeterminate long"></div>
                    <div class="v-progress-linear__indeterminate short"></div>
                  </div>
                </transition-stub>
                <!---->
              </div>
            </div>
            <!---->
            <div class="v-field__field" data-no-activator=""><label class="v-label v-field-label v-field-label--floating" aria-hidden="true" for="input-0">
                <!---->Valeur min
              </label><label class="v-label v-field-label" for="input-0">
                <!---->Valeur min
              </label>
              <!----><input size="1" type="number" id="input-0" aria-describedby="input-0-messages" inputmode="numeric" class="v-field__input">
              <!---->
            </div>
            <!---->
            <!---->
            <div class="v-field__outline">
              <!---->
              <!---->
            </div>
          </div>
        </div>
        <!---->
        <!---->
      </div>
      <div class="v-input v-input--horizontal v-input--center-affix v-input--density-default v-locale--is-ltr v-input--dirty v-text-field ma-3">
        <!---->
        <div class="v-input__control">
          <div class="v-field v-field--active v-field--center-affix v-field--dirty v-field--variant-filled v-theme--light v-locale--is-ltr">
            <div class="v-field__overlay"></div>
            <div class="v-field__loader">
              <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
                <!---->
                <div class="v-progress-linear__background" style="width: 100%;"></div>
                <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
                  <div class="v-progress-linear__indeterminate">
                    <div class="v-progress-linear__indeterminate long"></div>
                    <div class="v-progress-linear__indeterminate short"></div>
                  </div>
                </transition-stub>
                <!---->
              </div>
            </div>
            <!---->
            <div class="v-field__field" data-no-activator=""><label class="v-label v-field-label v-field-label--floating" aria-hidden="true" for="input-2">
                <!---->Valeur max
              </label><label class="v-label v-field-label" for="input-2">
                <!---->Valeur max
              </label>
              <!----><input size="1" type="number" id="input-2" aria-describedby="input-2-messages" ripple="false" inputmode="numeric" class="v-field__input">
              <!---->
            </div>
            <!---->
            <!---->
            <div class="v-field__outline">
              <!---->
              <!---->
            </div>
          </div>
        </div>
        <!---->
        <!---->
      </div>
    </div>
    <div class="v-input v-input--horizontal v-input--center-affix v-input--density-default v-locale--is-ltr v-input--dirty v-slider v-range-slider v-locale--is-ltr align-center mt-2 mb-6">
      <div class="v-input__prepend">
        <!---->0
        <!---->
      </div>
      <div class="v-input__control">
        <div class="v-slider__container"><input id="input-4_start" name="input-4" tabindex="-1"><input id="input-4_stop" name="input-4" tabindex="-1">
          <div class="v-slider-track" style="--v-slider-track-size: 4px; --v-slider-tick-size: 2px;">
            <div class="v-slider-track__background bg-primary v-slider-track__background--opacity" style="inset-inline-start: 0%; width: 100%;"></div>
            <div class="v-slider-track__fill bg-primary" style="inset-inline-start: 0%; width: 100%;"></div>
            <!---->
          </div>
          <div class="v-slider-thumb v-slider-thumb--focused v-locale--is-ltr" style="--v-slider-thumb-position: 0%; --v-slider-thumb-size: 20px;" role="slider" tabindex="0" aria-valuemin="0" aria-valuemax="6" aria-valuenow="0" aria-readonly="false" aria-orientation="horizontal" aria-describedby="input-4-messages">
            <div class="v-slider-thumb__surface text-primary elevation-2"></div>
            <div class="v-slider-thumb__ripple text-primary"></div>
            <transition-stub name="scale-transition" appear="false" persisted="false" css="true">
              <div class="v-slider-thumb__label-container" style="display: none;">
                <div class="v-slider-thumb__label">
                  <div>0</div>
                </div>
              </div>
            </transition-stub>
          </div>
          <div class="v-slider-thumb v-slider-thumb--focused v-locale--is-ltr" style="--v-slider-thumb-position: 100%; --v-slider-thumb-size: 20px;" role="slider" tabindex="0" aria-valuemin="0" aria-valuemax="6" aria-valuenow="6" aria-readonly="false" aria-orientation="horizontal" aria-describedby="input-4-messages">
            <div class="v-slider-thumb__surface text-primary elevation-2"></div>
            <div class="v-slider-thumb__ripple text-primary"></div>
            <transition-stub name="scale-transition" appear="false" persisted="false" css="true">
              <div class="v-slider-thumb__label-container" style="display: none;">
                <div class="v-slider-thumb__label">
                  <div>6</div>
                </div>
              </div>
            </transition-stub>
          </div>
        </div>
      </div>
      <div class="v-input__append">
        <!---->6
      </div>
      <!---->
    </div>
  </div>
</div>
`;
